tests/drivers/gnss/gnss_api/src/test_fix_rate.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gnss.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/ztest.h>
+
+#define TEST_FIX_TIMEOUT_MAX K_SECONDS(60)
+#define TEST_VALIDATE_PERIOD K_SECONDS(5)
+
+static const struct device *dev = DEVICE_DT_GET(DT_ALIAS(gnss));
+
+struct test_config {
+	uint32_t fix_interval;
+	uint32_t min_callback_count;
+	uint32_t max_callback_count;
+};
+
+static const struct test_config configs[] = {
+	{
+		.fix_interval = 100,
+		.min_callback_count = 40,
```

Comment:
Added macros to calculate the min/max and create test_config entries :) Upped the period to 10000ms to make callback counts evenly divisible up to 2000ms fix interval.

Reply:
Nice! Regarding your second point, alternatively the multiplication (`* 4` or `* 6`) could have been performed before the divison (`/ 5`) to avoid that issue altogether.

Reply:
Ah, thanks :)